ui/src/storage/storage.tsx

```diff
@@ -32,45 +30,58 @@ export const storeUserData: Middleware<unknown, RootState> =
     return result;
   };
 
-export function LoadingUserData(props: { children?: React.ReactNode }) {
-  const dispatch = useAppDispatch();
-  const status = useAsync<tanagra.UserData>(
-    useCallback(() => {
-      if (!storagePlugin) {
-        throw new Error("No storage plugin configured.");
-      }
+export async function fetchUserData(
+  dispatch: AppDispatch,
+  underlays: Underlay[]
+) {
+  if (!storagePlugin) {
+    throw new Error("No storage plugin configured.");
+  }
 
-      return storagePlugin.load().then((data?: tanagra.UserData) => {
-        if (!data) {
-          return;
-        }
+  const data = await storagePlugin.load();
+  if (!data) {
+    return;
+  }
 
-        if (data.version > currentVersion) {
-          throw new Error(
-            `Unable to load newer data: verson ${data.version} > current version ${currentVersion}`
-          );
-        }
+  if (data.version > currentVersion) {
+    throw new Error(
+      `Unable to load newer data: verson ${data.version} > current version ${currentVersion}`
+    );
+  }
 
-        // TODO(tjennison): Handle backwards compatability when we're closer to
-        // launch.
-        if (data.version != currentVersion) {
-          data.cohorts = [];
-          data.conceptSets = [];
-        }
+  // TODO(tjennison): Handle backwards compatability when we're closer to
+  // launch.
+  if (data.version != currentVersion) {
```

Comment:
This is checking if the cohort serialization is out of date with the current UI, yes? If the UI will always fetch these by calling core Tanagra endpoints, we should probably also include a check there.

Reply:
In theory it makes sense for that to live close to the data but we'll have to think through how all of the cases might work. Some examples:
- Can there be a divergence in version between the backend and UI?
- What about import?
- Could we need to display UI during an update?

Reply:
Sounds good, yeah we'll need to walk through difference scenarios (low priority for now). Tentatively, I'd hope we could always ship UI+backend together so we wouldn't need to handle different cohort schema versions.